### DISCUSSION TITLE
Elevation settings for zones

### DISCUSSION THREAD

**Discussion by sciurius** [home-assistant/frontend#13368] [Configuration] [answered]
Using the UI I can set an elevation which apparently applies to the home zone.
It is not possible to set elevations for individiual zones.

**Comment by ildar170975** [comment]
I was always interested what a purpose for "elevation"?

**Comment by sciurius** [comment]
I'd think for weather forecasts, sun rise/set times and such.

**Reply by matthiasdebaat** [reply]
Yes, what is your use case to have this for zones?

**Reply by dougiteixeira** [reply]
Just imagining... Would it be possible for the mobile application to obtain the information from the cell phone's barometer and create an entity? (the device tracker of the mobile device already brings the current altitude, but I believe it is based on the current coordinates).

Imagine being able to determine the area encompassing the coordinates and altitude to be able to determine if I am in my apartment or on the ground floor (technically I wouldn't be at home if I were outside the apartment, and currently by the coordinates this differentiation is not possible).

That would be a good reason for zones to have the altitude setting.

Examples:
Home Zone: XY coordinates with no configured altitude
Ground Floor Zone: XY coordinates with configured ZZZ altitude.

**Comment by sciurius** [comment]
Interesting question. If elevation plays a role in zone dependent data (e.g, a weather forecast), I'd expect this to be the elevation of the zone, not of the home zone.

**Reply by ildar170975** [reply]
Assume 2 persons live in a place with same geo coordinates.
Both persons have HA installed.
1st person specified 0 elevation, 2nd - 200 m.
Will they get different weather forecast ?))

**Comment by sciurius** [comment]
Let's spell it out, then.

1. Home Assistant has a setting for elevation.
2. This can only be set for the Home location. 
3. **If** elevation plays a role in location-dependant data, then it should be configurable for other zones as well.

**Reply by ildar170975** [reply]
Then the right question is - "what is a purpose of elevation for home zone?"

**Comment by matthiasdebaat** [comment, marked as answer]
As far as I know, elevation is only used for the core 'sun' integration, not for weather integrations. I suggested a documentation change for this https://github.com/home-assistant/home-assistant.io/pull/23773

The sun integration can only be used for your home location.

**Comment by sciurius** [comment]
Only sun, only home, okay.
